townhall/myapi/tests/test_volunteer.py

```diff
@@ -1,73 +1,124 @@
+        # Mocking the get_volunteer method before delete
+        mock_volunteer = townhall_models.Volunteer(
+            id=1, first_name="Zamorak", last_name="Red", email="zamorak.red@gmail.com", gender="M"
+        )
+        MockVolunteerServices.get_volunteer.return_value = mock_volunteer
 
-    def test_delete_volunteer(self):
         # Step 1
-        volunteer_1 = townhall_services.VolunteerServices.get_volunteer(id=1)
+        volunteer_1 = MockVolunteerServices.get_volunteer(id=1)
         assert volunteer_1.first_name == "Zamorak"
         assert volunteer_1.last_name == "Red"
         assert volunteer_1.email == "zamorak.red@gmail.com"
         assert volunteer_1.gender == "M"
+        MockVolunteerServices.get_volunteer.assert_called_once_with(id=1)
 
-        # Step 2
-        townhall_services.VolunteerServices.delete_volunteer(id=1)
-        assert townhall_services.VolunteerServices.get_volunteer(id=1) is None
+        # Mocking the delete_volunteer method
+        MockVolunteerServices.delete_volunteer.return_value = None
+        MockVolunteerServices.delete_volunteer(id=1)
+        MockVolunteerServices.delete_volunteer.assert_called_once_with(id=1)
 
-    # Retrieving all volunteers
-    def test_get_all_volunteers(self):
-        volunteers = townhall_services.VolunteerServices.get_volunteers_all()
+        # Mocking the get_volunteer method after delete
+        MockVolunteerServices.get_volunteer.return_value = None
+        assert MockVolunteerServices.get_volunteer(id=1) is None
+        MockVolunteerServices.get_volunteer.assert_called_with(id=1)
+
+        # Retrieving all volunteers
+    def test_get_all_volunteers(self, MockVolunteerServices):
+        # Mocking the get_volunteers_all method
+        MockVolunteerServices.get_volunteers_all.return_value = [
+            townhall_models.Volunteer(id=1, first_name="Zamorak", last_name="Red", email="zamorak.red@gmail.com", gender="M"),
+            townhall_models.Volunteer(id=2, first_name="Guthix", last_name="Green", email="guthix.green@hotmail.ca", gender="F")
+        ]
+
+        volunteers = MockVolunteerServices.get_volunteers_all()
         assert len(volunteers) == 2, "There should be two volunteers"
         assert volunteers[0].first_name == "Zamorak"
         assert volunteers[1].first_name == "Guthix"
+        MockVolunteerServices.get_volunteers_all.assert_called_once()
 
     # Retrieving all volunteers when there are none
-    def test_get_all_volunteers_empty(self):
-        # Clear all volunteers
-        townhall_models.Volunteer.objects.all().delete()
-
-        volunteers = townhall_services.VolunteerServices.get_volunteers_all()
-        assert len(volunteers) == 0, "There should be no volunteers"
+    def test_get_all_volunteers_empty(self, MockVolunteerServices):
+        
+        # Mocking the get_volunteers_all method for an empty result
+        MockVolunteerServices.get_volunteers_all.return_value = []
```

Comment:
This is me getting picky but because someone has gotten picky to me before, I want to impart this coding standard for improved readability, there is what is called[ Triple AAA testing principles that is considered Arrange Act Assert](https://medium.com/@pjbgf/title-testing-code-ocd-and-the-aaa-pattern-df453975ab80), which is just a fancy way of saying just put blank lines in test methods separating the 3 sections. Learning opportunity here too  @RYeeAnn @GagenvirGill @JordanBandur  :)